containers-tests/tests/intset-properties.hs

```diff
@@ -337,6 +337,11 @@ prop_foldL' s = foldl' (flip (:)) [] s == List.foldl' (flip (:)) [] (toList s)
 prop_map :: IntSet -> Bool
 prop_map s = map id s == s
 
+prop_mapMonotonic :: (Key -> Key) -> IntSet -> Property
+prop_mapMonotonic f s =
+  (\x y -> x < y ==> f x < f y) .&&.
+  mapMonotonic f s === map f s
```

Comment:
This is going to be too inefficient; too many test cases will be rejected. Also, properties should not take actual functions as arguments. QuickCheck offers a `Fun` type with growing and shrinking. I will have to think a bit about how to write this property optimally. For now, it might be okay to use the test you had before as long as you add a comment in the `mapMonotonic` implementation indicating that the test is insufficient to support any fancier implementation.

Reply:
I guess I can still do better than I originally did - I break down properties into 2: one to make sure it works correctly with `id`, and another just construct `f x = a*x + b` with positive `a`. So this at least get arbitrary linear functions covered.